### PR TITLE
Dashboard reports page

### PR DESCRIPTION
**Pull request by lukieh2014**
Finished the reports download page of the dashboard. Users can see and download previous daily summary reports from this page, and also filter for a selected date or date range of reports. Referring to ticket #111.

### CODE REVIEW

File: dashboard/daily_reports/daily_sales_report_2024-12-05.pdf
Comment (ebradley12) [inline dashboard/daily_reports/daily_sales_report_2024-12-05.pdf:1]: can we add to .gitignore

**Review by ebradley12 — APPROVED** [review]
wonderful

File: dashboard/dashboard.py [hunk 1]
```diff
@@ -45,6 +65,56 @@ def report_download_page() -> None:
...
+        min_value=date(2024, 12, 5),  # Earliest selectable date
+        max_value=date.today()       # Latest selectable date
+    )
+
+    if isinstance(date_range, tuple):
+        if len(date_range) == 2:
+            start_date, end_date = date_range
+            st.write(f"Start Date: {start_date}")
+            st.write(f"End Date: {end_date}")
+        else:
```
[the remark this answers is not in the document]
Comment (lukieh2014) [reply dashboard/dashboard.py:83]: just checked and doesn't seem to be possible as i am using a default date range to begin with

File: dashboard/dashboard.py [hunk 2]
```diff
@@ -45,6 +65,56 @@ def report_download_page() -> None:
...
+            # allow user to select just one date
+            start_date = date_range[0]
+            end_date = date_range[0]
+            st.write(f"Selected Date: {start_date}")
+
+        s3 = boto3.client('s3', aws_access_key_id=environ.get(
+            "aws_access_key_id"), aws_secret_access_key=environ.get("aws_secret_access_key"))
+        downloaded_reports = download_reports_from_s3(
+            s3, 'c14-bandcamp-reports')
```
[the remark this answers is not in the document]
Comment (lukieh2014) [reply dashboard/dashboard.py]: good idea, just implemented

**Review by bentrzcinski — APPROVED** [review]
nice stuff

**Pull request merged**